### DISCUSSION TITLE
rheoTestFoam with VCM and RRM constitutive model

### DISCUSSION THREAD

**Discussion by xiaohsu2006** [fppimenta/rheoTool#71] [General]
Dear Francisco Pimenta
      I am trying to run rheoTestFoam with VCM and RRM constitutive model. But an error occured reporting that 
“Create time

Create mesh for time = 0

SIMPLE: No convergence criteria found

Reading constitutiveProperties

Reading field p

Reading field U

Reading/calculating face flux field phi

Selecting constitutive equation VCM

--> FOAM FATAL ERROR: 
Not implemented

    From function virtual Foam::tmp<Foam::GeometricField<double, Foam::fvPatchField, Foam::volMesh> > Foam::constitutiveEq::etaSThermo() const
    in file constitutiveEqs/constitutiveEq/constitutiveEq.H at line 159.

FOAM aborting

#0  Foam::error::printStack(Foam::Ostream&) at ??:?
#1  Foam::error::abort() at ??:?
#2  Foam::constitutiveEq::etaSThermo() const at ??:?
#3  Foam::constitutiveEq::tauTotal() const at ??:?
#4  Foam::constitutiveModel::tauTotal() const at ??:?
#5  ? at ??:?
#6  __libc_start_main in "/lib64/libc.so.6"
#7  ? at ??:?
core dumped
”
 I checked the case carefully and could not fixed the problem. I would appreciate it if you can give me some advice and an example case will be best. Thank you.
Cheng Sun

**Comment by fppimenta** [comment]
Yes, there is a known bug with etaSThermo for all models that do not redefine it, as the ones you mention. This will be fixed in a next release. Meanwhile, you can downgrade to version 4.1 (https://github.com/fppimenta/rheoTool/releases) to get it working.

**Reply by xiaohsu2006** [reply]
Thanks for your indication.
